src/unit_tests/os_auth/test_auth_validate.c

```diff
@@ -257,9 +272,76 @@ static void test_w_auth_validate_groups(void **state) {
     err = w_auth_validate_groups(EXISTENT_GROUP1","EXISTENT_GROUP2","UNKNOWN_GROUP, response);
     assert_int_equal(err, OS_INVALID);
     assert_string_equal(response, "ERROR: Invalid group: "UNKNOWN_GROUP"");
+}
+
+static void test_w_auth_replace_agent_force_disabled(void **state) {
+    w_err_t err;
+    keyentry key;
```

Comment:
We can create a keyentry_init() method and a keyentry_free() method. It will be more clear for these tests and useful for future tests.

Reply:
For doing this I suggest duplicating the strings for id name and ip->IP (And free them in keyentry_free). It will be more secure.

Reply:
The methods were created !